File: Min_Elem/Min.go
/*
    Author : Pakshal Shashikant Jain 
    Date : 16/07/2023
    Program : Accept N numbers from user and return the Smallest number.
              Input : N : 6
                      Elements : 85 66 3 66 93 88
              Output : 3 
*/

//Required Libraries
package main
import "fmt"

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Function To Find Minimum Number from Entered N Number
func Minimum(isize int,arr []int) int {
	var i int = 0;
	var imin int = 0;

	//Logic To Find
	imin = arr[0];
	for i = 0;i < isize;i++ {
		if arr[i] < imin {
			imin = arr[i];
		}
	}

	return imin;
}

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Main function
func main() {
	var isize int = 0;
	var arr []int;
	var i int = 0;
	var iret int = 0;

	fmt.Println("Jay Ganesh......");

	//Taking Input from user
	fmt.Println("Enter Number of Element You Want in array");
	fmt.Scan(&isize);

	//Handeling Conditions
	if (isize < 0) {
		isize = -isize;
	}

	//Allocating Memory To array 
	arr = make([]int,isize);

	//Inserting Elements in array 
	fmt.Println("Enter Elements in array ");
	for i = 0;i < isize;i++ {
		fmt.Println("Enter Element Number ",i + 1);
		fmt.Scan(&arr[i]);
	} 

	//Call To Minimum function
	iret = Minimum(isize,arr);
	
	//Printing Minimum Element From Entered N Numbers
	fmt.Println("Printing Minimum From  Entered N Numbers",iret);
}

File: Min_Elem/Min.c
/*
    Author : Pakshal Shashikant Jain 
    Date : 16/07/2023
    Program : Accept N numbers from user and return the Smallest number.
              Input : N : 6
                      Elements : 85 66 3 66 93 88
              Output : 3 
*/

//Required Libraries
#include<stdio.h>
#include<stdlib.h>

//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Function To Find Minimum Number from Entered N Numbers
int Minimum(int isize,int arr[])
{
    int i = 0;
    int imin = 0;

    //Logic
    imin = arr[0];
    for(i = 0;i < isize;i++)
    {
        if(arr[i] < imin)
        {
            imin = arr[i];
        }
    }

    return imin;
}

/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

//Main function
int main()
{
    int isize = 0;
    int *arr = NULL;
    int i = 0;
    int iret = 0;

    //Taking Input from user
    printf("Enter Number of elements You Want in array \n");
    scanf("%d",&isize);

    //Handeling Conditions 
    if(isize < 0)
    {
        isize = -isize;
    }

    //Allocating Memory To array 
    arr = (int *)malloc(isize * sizeof(int));

    //Handeling Conditions Memory Not Allocated Properly 
    if(arr == NULL)
    {
        printf("Unable To Allocate Memory \n");
    
        return -1;
    }

    //Inserting Elements in array 
    printf("Enter Elements in array \n");
    for(i = 0;i < isize;i++)
    {
        printf("Enter Element number : %d\n",i + 1);
        scanf("%d",&arr[i]);
    }   

    //Call To Minimum function
    iret = Minimum(isize,arr);

    //Printing Minimum Number Found in Entered N Numbers
    printf("Minimum Number from Entered N Number is : %d",iret);

    free(arr);
    
    return 0;
}